### PR TITLE
added life module (with reading list):

### LINKED ISSUE #20: Reading list
Feature description:
-------------------
Feature to add a reading list, where the user can add articles, books, etc to their lists, and read whenever they want in the future.

### PR DESCRIPTION
**Pull request by systemplado**
#### Short description of what this resolves:
To solve and close #20 by making a `life` module with a `reading list`
#### Changes proposed in this pull request:
- added life path to `config.py`
- added life to `__init__.py`
- added `get_input` and `append_data_into_file`
- added `life` command in `yoda.py`
- changed indents to spaces in `diary.py`

### CODE REVIEW

**Comment by manparvesh** [comment]
Hi @systemplado sorry for late reply I got busy at work. 
I propose that we use `life` as a command group and `rlist` as its sub-command. Then the user will just need to write `yoda rlist` to do whatever he wants to. I suggest you take a look at the implementation of `vocabulary` if you want to see how this works.

**Comment by systemplado** [comment]
Alright working on it. Will update you soon as I finish it. I might add the `search` via `tags` (and hopefully I can extend the search by title and/or author)

File: modules/life.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            # click.echo("Read something? Enter id")
+
+            # key = int(get_input())
+            # if key in range(1, len(contents['entries'])+1):
+            #     article = contents['entries'][key-1]
+            #     read_from_rlist(article)
+            # else:
+            #     click.echo("That is why you fail.")
+    else:
+        empty_list_promt()
```
[the remark this answers is not in the document]
Comment (systemplado) [reply modules/life.py]: fixed https://github.com/yoda-pa/yoda/pull/22/commits/91ecfca5d25fa2630661f1ec56f9621e46e2d454#diff-46ff9ed8abaa9f87de72655cb5365445R101

**Comment by systemplado** [comment]
@manparvesh kindly check my newest commit https://github.com/yoda-pa/yoda/pull/22/commits/91ecfca5d25fa2630661f1ec56f9621e46e2d454

**Comment by manparvesh** [comment]
@systemplado I'll test this on my system as soon as I get time. Thanks for taking your time to contribute!

**Comment by systemplado** [comment]
I might remove the `path` in the reading list since the `open` functionality is not yet finished and I have no idea on how to test it on other OS.

**Comment by manparvesh** [comment]
1. I got an error while creating an entry, saying: `No such file or directory: '/home/mp/.yoda/life/reading_list.yaml'`. I think you didn't create the folder. There is a function for that in the util.py file. I think that'll help
2. Also, there some some changes made in the `config.py` file, so please make appropriate changes to the same.

**Comment by systemplado** [comment]
@manparvesh kindly check my PR. I've added the needed changes 😁. You might want to squash and merge, if ever you will approve, to avoid having multiple commits from this PR.

**Comment by manparvesh** [comment]
@systemplado looks cool to me! :smile: Thanks for contributing!  
I am merging this, but are still many things to be added in this module.

File: modules/life.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+@click.group()
+def life():
+    '''
+        Life command group:\n
+        contains helpful commands to organize your life
+    '''
+
+
+def reading_list_entry_file_path():
+    return LIFE_CONFIG_FOLDER_PATH + '/' + "reading_list.yaml"
```
Comment (himanshub16) [inline modules/life.py]: Hey,
Would recommend using `os.path.join` or `os.sep` to make the code platform agnostic.

Comment (systemplado) [reply modules/life.py]: Currently implementing it. Anything else you think that needs to be changed?

**Pull request merged**